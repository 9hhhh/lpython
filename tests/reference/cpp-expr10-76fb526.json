{
    "basename": "cpp-expr10-76fb526",
    "cmd": "lpython --no-color --show-cpp {infile}",
    "infile": "tests/expr10.py",
    "infile_hash": "f255430e3c0729835b5406c7341c20d3a826777f0e21bcdd1245445b",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "cpp-expr10-76fb526.stdout",
    "stdout_hash": "3d6bf0a13313460664bd2ce32c48416ad7034742d12adf67698d6c22",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}