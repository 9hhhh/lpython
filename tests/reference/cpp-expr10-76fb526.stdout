#include <iostream>
#include <string>
#include <vector>
#include <Kokkos_Core.hpp>
#include <lfortran_intrinsics.h>

template <typename T>
Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
{
    Kokkos::View<T*> r("r", v.size());
    for (size_t i=0; i < v.size(); i++) {
        r(i) = v[i];
    }
    return r;
}

void test_UnaryOp()
{
    int a;
    bool b1;
    bool b2;
    float f;
    a = 4;
    a = -500;
    f = 1.000000;
    f = -183745.534000;
    b1 = true;
    b2 = !false;
}

